core/dbt/parser/schemas.py

```diff
@@ -399,8 +399,13 @@ def generate_source_node(self, source, table, path, package_name, root_dir,
 
         loaded_at_field = table.get('loaded_at_field',
                                     source.get('loaded_at_field'))
+        default_database = self.root_project_config.credentials.database
+        default_schema = self.root_project_config.credentials.schema
```

Comment:
i don't think this is the best value for a default schema -- it would be pretty unlikely if people had source tables in their dbt schemas! I think a sane default for the schema would be the `name` of the source, but if that's not doable, then i wouldn't bother with a default.

For the `identifier` field below -- does that default to the `name` of the source _table_? If not, I think that could be a really good thing to add. Well behaved datasets like Snowplow can be described with:
```
version: 2

sources:
  - name: snowplow
    loader: snowpipe
    tables:
      - name: event
```

If you're using stock Snowplow, then the effective relation name will just be `<database>.snowplow.event`, which is pretty great! The ability to supply custom overrides for the database, schema, and table identifier are all for the less well-behaved relations that eg. include illegal chars in their table names (which need to be quoted)

Reply:
^ on that topic, we should quote these Relations whenever we render them out! The casing supplied by the user should exactly match the value in the database